internal/controller/cilium_flow_manager.go

```diff
@@ -0,0 +1,228 @@
+package controller
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/cilium/cilium/api/v1/flow"
+	observer "github.com/cilium/cilium/api/v1/observer"
+	pb "github.com/illumio/cloud-operator/api/illumio/cloud/k8scluster/v1"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// FlowManager holds a logger and Cilium client
+type FlowManager struct {
+	logger *zap.SugaredLogger
+	client observer.ObserverClient
+}
+
+// discoverHubbleRelayAddress uses a kubernetes clientset in order to discover the address of hubble-relay within kube-system.
+func discoverHubbleRelayAddress(ctx context.Context, logger *zap.SugaredLogger, clientset kubernetes.Interface) (string, error) {
+	service, err := clientset.CoreV1().Services("kube-system").Get(ctx, "hubble-relay", metav1.GetOptions{})
+	if err != nil {
+		logger.Errorw("Failed to get Hubble Relay service", "error", err)
+		return "", err
```

Comment:
This log is redundant with that of the caller (`BootUpStreamAndReconnect`). Just return a more descriptive error, and don't log in this method.
```suggestion
		return "", fmt.Errorf("Cilium Hubble Relay service not found: %w", err)
```

Reply:
Done